at_client/src/main/java/org/atsign/client/api/impl/clients/AtClientImpl.java

```diff
@@ -410,8 +414,52 @@ private String _delete(PublicKey key) {
     private String _put(SelfKey selfKey, byte[] value) throws AtException {throw new RuntimeException("Not Implemented");}
     private String _put(PublicKey publicKey, byte[] value) throws AtException {throw new RuntimeException("Not Implemented");}
 
-    private List<AtKey> _getAtKeys(String regex) {
-        throw new RuntimeException("Not Implemented");
+    private List<AtKey> _getAtKeys(String regex) throws AtException {
+        Secondary.Response rawResponse = executeCommand("scan", false);
+        String[] rawArray = StringUtil.getRawStringArrayFromScanRawResponseString(rawResponse.data);
```

Comment:
As per discussion just now, please decode using the Jackson ObjectMapper something like 
```
ObjectMapper mapper = new ObjectMapper();
try {
    List<String> strings = mapper.readerForListOf(String.class).readValue(value.data);
}
``` etc